core/src/main/java/com/datastrato/gravitino/lock/LockManager.java

```diff
@@ -0,0 +1,195 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * LockManager is a lock manager that manages the tree locks. It will service as a factory to create
+ * the tree lock and do the cleanup for the stale tree locks node.
+ */
+public class LockManager {
+  private static final Logger LOG = LoggerFactory.getLogger(LockManager.class);
+
+  static final NameIdentifier ROOT = NameIdentifier.of("/");
+
+  @VisibleForTesting TreeLockNode treeLockRootNode;
+  final AtomicLong totalNodeCount = new AtomicLong(1);
+
+  // The maximum number of tree lock nodes to keep in memory. If the total node count is greater
+  // than this value, we will do the cleanup.
+  @VisibleForTesting long maxTreeNodeInMemory;
+  // If the total node count is less than this value, we will not do the cleanup.
+  @VisibleForTesting long minTreeNodeInMemory;
+
+  // The interval in seconds to clean up the stale tree lock nodes.
+  @VisibleForTesting long cleanTreeNodeIntervalInSecs;
+
+  private void initParameters(Config config) {
+    long maxNodesInMemory = config.get(TREE_LOCK_MAX_NODE_IN_MEMORY);
+    if (maxNodesInMemory <= 0) {
+      throw new IllegalArgumentException(
+          String.format(
+              "The maximum number of tree lock nodes '%d' should be greater than 0",
+              maxNodesInMemory));
+    }
+
+    long minNodesInMemory = config.get(TREE_LOCK_MIN_NODE_IN_MEMORY);
+    if (minNodesInMemory <= 0) {
+      throw new IllegalArgumentException(
+          String.format(
+              "The minimum number of tree lock nodes '%d' should be greater than 0",
+              minNodesInMemory));
+    }
+
+    if (maxNodesInMemory <= minNodesInMemory) {
+      throw new IllegalArgumentException(
+          String.format(
+              "The maximum number of tree lock nodes '%d' should be greater than the minimum number of tree lock nodes '%d'",
+              maxNodesInMemory, minNodesInMemory));
+    }
+    this.maxTreeNodeInMemory = maxNodesInMemory;
+    this.minTreeNodeInMemory = minNodesInMemory;
+
+    long cleanIntervalInSecs = config.get(TREE_LOCK_CLEAN_INTERVAL);
+    if (cleanIntervalInSecs <= 0) {
+      throw new IllegalArgumentException(
+          String.format(
+              "The interval in seconds to clean up the stale tree lock nodes '%d' should be greater than 0",
+              cleanIntervalInSecs));
+    }
+
+    this.cleanTreeNodeIntervalInSecs = cleanIntervalInSecs;
+  }
+
+  private void startNodeCleaner() {
+    ScheduledThreadPoolExecutor lockCleaner =
+        new ScheduledThreadPoolExecutor(
+            1,
+            new ThreadFactoryBuilder()
+                .setDaemon(true)
+                .setNameFormat("tree-lock-cleaner-%d")
+                .build());
+
+    lockCleaner.scheduleAtFixedRate(
+        () -> {
+          long nodeCount = totalNodeCount.get();
+          LOG.trace("Total tree lock node count: {}", nodeCount);
+          if (nodeCount > maxTreeNodeInMemory) {
```

Comment:
I think start cleanup when the nodeCount> maxTreeNodeInMemory/2.
We should handle  the nodeCound > maxTreeNodeInMemory，it maybe cause out of memory